contracts/contracts/NoLossDao.sol

```diff
@@ -141,23 +151,26 @@ contract NoLossDao is Initializable {
     );
     _;
   }
+
+  modifier depositContractOnly() {
+    require(
+      address(depositContract) == msg.sender, // Is this a valid way of getting the address?
+      'function can only be called by deposit contract'
+    );
+    _;
+  }
+
   ////////////////////////////////////
   //////// SETUP CONTRACT////////////
   //// NOTE: Upgradable at the moment
   function initialize(
     address daiAddress,
-    address aDaiAddress,
-    address aavePoolAddress,
-    address aavePoolCoreAddress,
-    uint256 _proposalAmount,
+    address _depositContractAddress,
     uint256 _votingInterval
   ) public initializer {
```

Comment:
Almost all of our functions should be converted to using `external`. You only need to use public if the function will be called internally too (you cannot call external functions internally).

We've been sloppy with this on other code we have, but it isn't best practice...

https://ethereum.stackexchange.com/a/19391/4642 (the second answer is also insightful)